### DISCUSSION TITLE
Allow passing paramenters from handlers to bot session middleware

### DISCUSSION THREAD

**Discussion by torikki-tou** [aiogram/aiogram#1487] [Ideas]
I can populate request context in dispatcher middleware, can use data from that context in my handler but i completely lose it in my request middleware in my bot although it is still part of my event dispatching workflow

Code example:
```python
async def update_middleware(
        handler: Callable[[aiogram.types.Update, Dict[str, Any]], Awaitable[Any]],
        event: aiogram.types.Update,
        data: Dict[str, Any]
):
    data['trace_id'] = "someTraceID"
    return await handler(event, data)


@router.message()
async def handle(message: aiogram.types.Message, trace_id: str):
    await message.answer(message.text)  # i want to pass trace_id here


async def request_middleware(
        make_request: NextRequestMiddlewareType[TelegramType],
        bot: aiogram.Bot,
        method: TelegramMethod[TelegramType],
) -> Response[TelegramType]:
    # i want to receive trace_id here
    pass


bot = aiogram.Bot(token="xxx")
bot.session.middleware.register(request_middleware)

dispatcher = aiogram.Dispatcher()
dispatcher.update.outer_middleware.register(update_middleware)
```

Am i missing something? Or is it intended?

**Comment by JrooTJunior** [comment]
Completely impossible to get data from the event middlewares inside client middleware because it wors on diametrically opposite sides of the application.

At the same time you can use contextvars to get contextual data inside request handler if it is available at the moment.

Note that the API methods (and client session middlewares) can be called outside of the handlers and event middlewares (like getUpdates and getMe methods is called before other scopes can be caused)

**Reply by torikki-tou** [reply]
Can't we just use `**kwargs` and unpack it in the request middleware?

Kinda like that
```python
@router.message()
async def handle(message: aiogram.types.Message, trace_id: str):
    kwargs["trace_id"] = trace_id
    await message.answer(message.text)  


async def request_middleware(
        make_request: NextRequestMiddlewareType[TelegramType],
        bot: aiogram.Bot,
        method: TelegramMethod[TelegramType],
       **kwargs,
) -> Response[TelegramType]:
    trace_id = kwargs.get("trace_id")
    pass
```

> it wors on diametrically opposite sides of the application

Recreate same mechanism on that other side?
